app/src/main/java/com/zeapo/pwdstore/UserPreference.kt

```diff
@@ -519,7 +520,24 @@ class UserPreference : AppCompatActivity() {
 
     @Throws(IOException::class)
     private fun copySshKey(uri: Uri) {
-        // TODO: Check if valid SSH Key before import
+        // See metadata from document to validate SSH key
+        contentResolver.query(
+                uri, null, null, null, null, null
+        )?.use { cursor ->
+            val sizeIndex: Int = cursor.getColumnIndex(OpenableColumns.SIZE)
+
+            // cursor returns only 1 row
+            cursor.moveToFirst()
+            // see file's metadata
+            val sizeFile: Int = cursor.getInt(sizeIndex)
+            val extensionFile = File(uri.path.toString()).extension
+
+            // SSH key without file extension and size < than 100KB
+            if (extensionFile.isNotEmpty() || //
```

Comment:
I would personally remove the check for the file extension and simply keep the size check. After all, loading huge files and having APS choke on them is what we want to prevent, everything else is both much harder to prevent and less problematic.